lib/cext/preprocess.rb

```diff
@@ -30,7 +35,44 @@
 
       "#{shim}(#{argc}, #{argv}, \"#{arity}\", "
     }
+  elsif line =~ /^(\s+)VALUE((?:\s+\w+\s*(?:\[\s*\d+\s*\]\s*)?,?)+);\s*$/
```

Comment:
I think the second `\s` needs to be qualified with `*`. Could you check by adding a test for `VALUE a,b[2],c;` ?